plugin/src/main/java/org/jenkinsci/plugins/structs/describable/DescribableModel.java

```diff
@@ -267,12 +271,20 @@ public String getDisplayName() {
      * or {@link Class#getSimpleName} may be used in case the argument type is {@link Describable}
      * and only one subtype is registered (as a {@link Descriptor}) with that simple name.
      */
-    public T instantiate(Map<String,?> arguments) throws Exception {
+    public T instantiate(Map<String,?> arguments) throws IllegalArgumentException {
+        return instantiate(arguments, new LogTaskListener(LOGGER, Level.FINE));
+    }
+
+    public T instantiate(Map<String,?> arguments, TaskListener listener) throws IllegalArgumentException {
+        if (listener == null) {
+            listener = new LogTaskListener(LOGGER, Level.FINE);
+        }
+
         CustomDescribableModel cdm = CustomDescribableModel.of(type);
         if (cdm != null) {
             Map<String, Object> input = deeplyImmutable(arguments);
             arguments = cdm.customInstantiate(input);
-            LOGGER.log(Level.FINE, "{0} translated {1} to {2}", new Object[] {cdm.getClass(), input, arguments});
+            listener.getLogger().format("%s translated %s to %s", cdm.getClass(), input, arguments);
```

Comment:
I think it would be best to leave this using `LOGGER`, it looks like more of a debugging aid for developers implementing `CustomDescribableModel` than something that would be useful to users.

Reply:
Yes, we expect all this to be silent unless something is wrong.